frontend/src/app/offers/create-offer/create-offer.component.ts

```diff
@@ -0,0 +1,196 @@
+      this.offer = response as AppOffer;
+      this.inEditMode = true;
+      this.offer = this.removeTime();
+      this.form.patchValue(this.offer);
+    });
+  }
+
+// Temporary fix to make editing work properly, to delete after we fix datetime issue
+  removeTime() {
+    this.offer.startedAt = this.offer.startedAt.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    this.offer.finishedAt = this.offer.finishedAt.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    if (this.offer.recruitmentEndDate) {
+      this.offer.recruitmentEndDate = this.offer.recruitmentEndDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    };
+    if (this.offer.recruitmentStartDate) {
+      this.offer.recruitmentStartDate = this.offer.recruitmentStartDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    };
+    if (this.offer.reserveRecruitmentEndDate) {
+      this.offer.reserveRecruitmentEndDate = this.offer.reserveRecruitmentEndDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    }
+    if (this.offer.reserveRecruitmentStartDate) {
+      this.offer.reserveRecruitmentStartDate = this.offer.reserveRecruitmentStartDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    }
+    return this.offer;
+  }
+
+  // TODO - delete when we decide what date format we want to have
+  addTime() {
+    this.form.value.startedAt = this.form.value.startedAt + 'T00:00:00Z';
+    this.form.value.finishedAt = this.form.value.finishedAt + 'T00:00:00Z';
+
+    if (this.form.value.recruitmentEndDate) {
+      this.form.value.recruitmentEndDate = this.form.value.recruitmentEndDate + 'T00:00:00Z';
+    }
+    if (this.form.value.recruitmentStartDate) {
+      this.form.value.recruitmentStartDate = this.form.value.recruitmentStartDate + 'T00:00:00Z';
+    }
+    if (this.form.value.reserveRecruitmentEndDate) {
+      this.form.value.reserveRecruitmentEndDate = this.form.value.reserveRecruitmentEndDate + 'T00:00:00Z';
+    }
+    if (this.form.value.reserveRecruitmentStartDate) {
+      this.form.value.reserveRecruitmentStartDate = this.form.value.reserveRecruitmentStartDate + 'T00:00:00Z';
+    }
+  }
+
+  ngOnDestroy() {
+    if (this.userSubscription) {
+      this.userSubscription.unsubscribe();
+    };
+  }
+
+  toDataUrl(url, callback) {
+    const reader = new FileReader();
+    this.http.get(url, {responseType: 'blob'})
+    .subscribe(response => {
+      reader.onloadend = function() {
+          callback(reader.result)
+      }
+      reader.readAsDataURL(response);
+    })
+  };
+
+  onSubmit(offer: AppOffer) {
+    if (this.form.valid) {
+      this.addTime()
+    }
+    if (this.offer.image) {
+      this.form.value.image = { ...this.offer.image };
+      this.form.value.image.content = this.form.value.image.content.replace(/.*,/, '');
+    }
+    if (this.inEditMode) {
+      this.offersService.editOffer(this.form.value, offer.id)
+      .subscribe(
+        response => {
+        if (response.status === 200) {
```

Comment:
We don't need to check the status here, as all success statuses ( from 200 to 299) resolve here, so all logic in this function is executed when the request was successfull, hence we don't need to observe the response and status if we care only if it is successfull, and I guest that's what we are doing here.